ql/src/java/org/apache/hadoop/hive/ql/stats/BasicStatsNoJobTask.java

```diff
@@ -119,16 +129,83 @@ public String getName() {
     return "STATS-NO-JOB";
   }
 
-  static class StatItem {
-    Partish partish;
-    Map<String, String> params;
-    Object result;
+  abstract static class StatCollector implements Runnable {
+
+    protected Partish partish;
+    protected Object result;
+    protected LogHelper console;
+
+    public static Function<StatCollector, String> SIMPLE_NAME_FUNCTION =
+        sc -> String.format("%s#%s", sc.partish().getTable().getCompleteName(), sc.partish().getPartishType());
+
+    public static Function<StatCollector, Partition> EXTRACT_RESULT_FUNCTION = sc -> (Partition) sc.result();
+
+    abstract Partish partish();
+    abstract boolean isValid();
+    abstract Object result();
+    abstract void init(HiveConf conf, LogHelper console) throws IOException;
+
+    protected String toString(Map<String, String> parameters) {
+      return StatsSetupConst.SUPPORTED_STATS.stream().map(st -> st + "=" + parameters.get(st))
+          .collect(Collectors.joining(", "));
+    }
   }
 
-  static class FooterStatCollector implements Runnable {
+  static class HiveStorageHandlerStatCollector extends StatCollector {
+
+    public HiveStorageHandlerStatCollector(Partish partish) {
+      this.partish = partish;
+    }
+
+    @Override
+    public void init(HiveConf conf, LogHelper console) throws IOException {
+      this.console = console;
+    }
+
+    @Override
+    public void run() {
+      try {
+        Table table = partish.getTable();
+        Map<String, String> parameters = partish.getPartParameters();
+        TableDesc tableDesc = Utilities.getTableDesc(table);
+        Map<String, String> basicStatistics = table.getStorageHandler().getBasicStatistics(tableDesc);
+
+        StatsSetupConst.setBasicStatsState(parameters, StatsSetupConst.TRUE);
```

Comment:
I don't understand why we make changes to the `Table` when we could be updating infos of a partition as well...
I guess in case of IceBerg you will not have regular partitions ; so it will probably work for that correctly

I think here you want to change `parameters`